packs/chef/actions/lib/shell_out.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python2.7
+
+import sys
+import os
+import subprocess
+import select
+
+# Starts command in subprocess processing output as it appears.
+def shell_out(command, env={}):
+    env = os.environ.copy().update(env)
+    proc = subprocess.Popen(args=command, stdin=None, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, env=env)
+    while True:
+        fdlist = [proc.stdout.fileno(), proc.stderr.fileno()]
+        ios = select.select(fdlist, [], [])
+        for fd in ios[0]:
+            if fd == proc.stdout.fileno():
+                sys.stdout.write(proc.stdout.read())
```

Comment:
Tomaz, readline() is a bad idea also as it happened. http://gyazo.com/6a954c05aa2d75b72c799205028b5d31
read(io.DEFAULT_BUFFER_SIZE) doesn't harass CPU and it's about ~ 9%. read(n) call is non-blocking it well return less bytes if n available.
About common utils it's cool, agreed. But what if it's a remote script runner on a remote host without st2 pre-installed?